controllers/spaceprovisionerconfig/spaceprovisionerconfig_controller.go

```diff
@@ -58,62 +74,183 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
+		updateReadyCondition(spc, corev1.ConditionFalse, toolchainv1alpha1.SpaceProvisionerConfigDisabledReason, "")
+		return nil
+	}
+
+	clusterCondition, err := r.determineClusterReadyState(ctx, spc)
+	if err != nil {
+		updateReadyCondition(spc, clusterCondition, toolchainv1alpha1.SpaceProvisionerConfigToolchainClusterNotFoundReason, err.Error())
+
+		// the reconciler reacts on ToolchainCluster changes so it will be triggered once a new TC appears
+		// we therefore don't need to return error from the reconciler in the case the TC is not found.
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+
+	var reason string
+	resultCondition := clusterCondition
+
+	if clusterCondition == corev1.ConditionTrue {
+		collectUsage := r.GetUsageFunc
+		if collectUsage == nil {
+			collectUsage = collectConsumedCapacity
 		}
-		reportedError = fmt.Errorf("failed to update the SpaceProvisionerConfig status: %w", err)
+		cc, err := collectUsage(ctx, r.Client, spc.Spec.ToolchainCluster, spc.Namespace)
+		if err != nil {
+			updateReadyCondition(spc, corev1.ConditionUnknown, toolchainv1alpha1.SpaceProvisionerConfigFailedToDetermineCapacityReason, err.Error())
+			return err
+		}
+		spc.Status.ConsumedCapacity = cc
+
+		capacityCondition := r.determineCapacityReadyState(spc)
+		if capacityCondition != corev1.ConditionTrue {
+			reason = toolchainv1alpha1.SpaceProvisionerConfigInsufficientCapacityReason
+		} else {
+			reason = toolchainv1alpha1.SpaceProvisionerConfigValidReason
+		}
+
+		resultCondition = capacityCondition
+	} else {
+		reason = toolchainv1alpha1.SpaceProvisionerConfigToolchainClusterNotReadyReason
+	}
+
+	readyCondition := toolchainv1alpha1.Condition{
+		Type:   toolchainv1alpha1.ConditionReady,
+		Status: resultCondition,
+		Reason: reason,
 	}
 
-	return ctrl.Result{}, reportedError
+	spc.Status.Conditions, _ = condition.AddOrUpdateStatusConditions(spc.Status.Conditions, readyCondition)
+
+	return nil
 }
 
-func (r *Reconciler) determineReadyState(ctx context.Context, spc *toolchainv1alpha1.SpaceProvisionerConfig) (toolchainv1alpha1.Condition, error) {
+// Note that this function merely mirrors the usage information found in the ToolchainStatus. This means that it actually may work
+// with slightly stale data because the counter.Counts cache might not have been synced yet. This is ok though because the capacity manager
+// doesn't completely rely on the readiness status of the SPC and will re-evaluate the decision taking into the account the contents of
+// the counter cache and therefore completely "fresh" data.
+func collectConsumedCapacity(ctx context.Context, cl runtimeclient.Client, clusterName string, toolchainStatusNs string) (*toolchainv1alpha1.ConsumedCapacity, error) {
+	status := &toolchainv1alpha1.ToolchainStatus{}
+	if err := cl.Get(ctx, types.NamespacedName{Namespace: toolchainStatusNs, Name: toolchainconfig.ToolchainStatusName}, status); err != nil {
+		if errors.IsNotFound(err) {
+			return nil, nil
```

Comment:
You're not missing anything I think :) Returning `nil` here has essentially the same effect - the readiness status will flip to `Unknown`, just the reason would be different. I changed it to return an error here, because the reason would actually be more meaningful. Thanks for spotting it.